ast/expressions_test.go

```diff
@@ -155,3 +155,44 @@ func (tc *testExpressionsSuite) TestBetweenExprRestore(c *C) {
 	}
 	RunNodeRestoreTest(c, testCases, "select %s", extractNodeFunc)
 }
+
+func (tc *testExpressionsSuite) TestCaseExpr(c *C) {
+	testCases := []NodeRestoreTestCase{
+		{"case when 1 then 2 end", "CASE WHEN 1 THEN 2 END"},
+		{"case when 1 then 'a' when 2 then 'b' end", "CASE WHEN 1 THEN 'a' WHEN 2 THEN 'b' END"},
+		{"case when 1 then 'a' when 2 then 'b' else 'c' end", "CASE WHEN 1 THEN 'a' WHEN 2 THEN 'b' ELSE 'c' END"},
+		{"case when 'a'!=1 then true else false end", "CASE WHEN 'a'!=1 THEN TRUE ELSE FALSE END"},
+		{"case a when 'a' then true else false end", "CASE `a` WHEN 'a' THEN TRUE ELSE FALSE END"},
+	}
+	extractNodeFunc := func(node Node) Node {
+		return node.(*SelectStmt).Fields.Fields[0].Expr
+	}
+	RunNodeRestoreTest(c, testCases, "select %s", extractNodeFunc)
+}
+
+func (tc *testExpressionsSuite) TestBinaryOperationExpr(c *C) {
+	testCases := [] NodeRestoreTestCase{
+		{"'a'!=1", "'a'!=1"},
+		{"a!=1", "`a`!=1"},
```

Comment:
Please add a test case for `a <> 1`.

Reply:
I've changed as your suggests
@kennytm PTAL